### PR TITLE
CI: Acceptance tests at PR: add build containers task

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
## What does this PR change?

In order to run acceptance tests at PR we need to build the containers:

* opensuse-minion: a container that simulates an opensuse minion
* rocky-minion: a container that simulates a rocky minion
* ubuntu-minion: a container that simulates an ubuntu minion
* server-all-in-one: a container that simulates a server
* server-all-in-one-dev: a container, based on server-all-in-one, which also contains the build dependencies for building new code
* controller: a container which we will use to run the cucumber test tasks

This commit adds Dockerfiles for all of them an a github action workflow that can be used to trigger the build of it.

## GUI diff

N/A

- [X] **DONE**

## Documentation
- No documentation needed
- [X] **DONE**

## Test coverage
- No tests
- [X] **DONE**

## Links

Related to https://github.com/SUSE/spacewalk/issues/20680

- [X] **DONE**

## Changelogs

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by srbarrios — APPROVED** [review]
lgtm

**Comment by cbosdo** [comment]
> About https://github.com/uyuni-project/uyuni/tree/server-container/containers/server-image: sure we should use that one! Any plans to publish that one in some registry (ghcr or registry.opensuse.org)?

It is already built in https://build.opensuse.org/package/show/systemsmanagement:Uyuni:Master:ServerContainer/server-image though that is not an official place yet. You can switch to it once we merge that branch in master.

**Comment by cbosdo** [comment]
Looks like a rebase on latest master would clear the sonarcloud issues.

**Comment by jordimassaguerpla** [comment]
> Is there really no other way than adding `minima` binary to the git repo? Can't it be pulled or built from sources inside a container?
> 
> Could you please remove the root password setting in the `server-all-in-one` image? This is not used, so better not set it.
> 
> In the short term future, it would be cool to merge that image with the one we build in the [server-container branch](https://github.com/uyuni-project/uyuni/tree/server-container/containers/server-image).

Thanks!

I will look into it. Definitely adding the minima binary is not a good "practice" ... I will see what I can do.

I will remove the root password.

About https://github.com/uyuni-project/uyuni/tree/server-container/containers/server-image: sure we should use that one! Any plans to publish that one in some registry (ghcr or registry.opensuse.org)?


> > About https://github.com/uyuni-project/uyuni/tree/server-container/containers/server-image: sure we should use that one! Any plans to publish that one in some registry (ghcr or registry.opensuse.org)?
> 
> It is already built in https://build.opensuse.org/package/show/systemsmanagement:Uyuni:Master:ServerContainer/server-image though that is not an official place yet. You can switch to it once we merge that branch in master.

We would need a container that is stable, so that between 2 runs of the github action would produce the same result. Using a container that is continuously being built could produce different results.

Do we plan to push this to some stable repository?

**Comment by jordimassaguerpla** [comment]
@cbosdo could you take a second look? I have removed the root password and I get the minima tool from github repo, the same way we do in our internal infrastructure.

Until we have a stable repo for the server container, I prefer to use the one in github. This is anyway for the beta, so people can start using it, but we can create a github issue about it and make sure we fix it later. Deal?

File: testsuite/dockerfiles/server-all-in-one/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+RUN zypper ar -G http://download.opensuse.org/update/leap/15.4/oss/ os_update_repo
+RUN zypper ar -G http://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable/images/repo/Uyuni-Server-POOL-x86_64-Media1/ server_pool_repo
+RUN zypper ar -G http://download.opensuse.org/update/leap/15.4/sle/ sle_update_repo
+RUN zypper ar -G http://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable/images/repo/Testing-Overlay-POOL-x86_64-Media1/ testing_overlay_devel_repo
+
+copy setup_env.sh /root/setup_env.sh
+
+RUN zypper ref
+RUN zypper -n in vi
+RUN zypper -n in python3-simplejson
```
Comment (cbosdo) [inline testsuite/dockerfiles/server-all-in-one/Dockerfile:15]: `python3-simplejson` is no longer needed

**Review by cbosdo — APPROVED** [review]
LGTM. Of course using the same image can wait.

**Pull request merged**